llm/server.go

```diff
@@ -0,0 +1,643 @@
+package llm
```

Comment:
Much of this file is from https://github.com/ollama/ollama/blob/v0.1.17/llm/llama.go